File: Data Structures & Algorithms/maximum-subarray/submission-3.java
class Solution {
    public int maxSubArray(int[] nums) {
        if(nums.length==1) return nums[0];
        if(nums.length==2) return Math.max(nums[0],nums[1]);
        int prevSum =nums[0],res=nums[0];
        int i=0;
        for(int j=1;j<nums.length;j++){
            if(nums[j]+prevSum>=0){ 
                prevSum+=nums[j];
                res = Math.max(res,prevSum);
                }
            else{
                res = Math.max(res,nums[j]);
                prevSum=0;
            } 
        }
        return res;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s)==0:
            return 0
        if len(s)==1:
            return 1
        l,r=0,1
        res=0
        sett=set(s[0])
        while r<len(s) and l<len(s):
            while s[r] in sett:
                sett.remove(s[l])
                l+=1
            sett.add(s[r])
            r+=1
            res = max(res,r-l+1)
        return res-1
            
        